src/MOI_wrapper/indicator_constraints.jl

```diff
@@ -0,0 +1,35 @@
+# Indicator constraints
+
+"""
+Set of (x,y) that satisfy the indicator constraint:
+
+    y ∈ 𝔹, y = 1 ==> a^T x ≤ rhs
+
+"""
+struct IndicatorSet{T <: Real} <: MOI.AbstractVectorSet
+    a::Vector{T}
+    rhs::T
+end
+
+MOI.supports_constraint(::Optimizer, ::Type{VECTOR}, ::Type{IS}) where {IS <: IndicatorSet{<:AbstractFloat}} = true
+
+function MOI.add_constraint(o::Optimizer, func::VECTOR, set::IndicatorSet{Float64})
+    length(func.variables) == length(set.a) + 1 || throw(DimensionMismatch("Indicator set and VectorOfVariables length mismatch"))
+    allow_modification(o)
+    y = VarRef(func.variables[1].value)
```

Comment:
This would be a nice place to check whether y is already binary.

Reply:
shouldn't we do it in the SCIP part instead of MOI? (meaning here in the parts defined in `managed_scip.jl`?)

Reply:
Either way is fine with me. I only looked up how I did it with SOC but can not remember the rationale. I guess I thought that the MOI part was more "user-facing", but of course if we move the check "down" the call stack, it will cover more cases!